components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import Dropdown from 'react-bootstrap/Dropdown'
+import UserInfoImage from './UserInfoImage'
+import { ChevronDownIcon } from '@primer/octicons-react'
+import LogoutContainer from './LogoutContainer'
+import { Button } from './theme/Button'
+import { UserInfo } from '../@types/user'
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+type CustomToggleProps = {
+  children?: React.ReactNode
+  className?: string | undefined
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {}
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+
+  const fullname = _.get(data, 'userInfo.user.name', '')
+
+  const userInfo: UserInfo = {
+    // 'A' stands for Anonymous, in case user did not put in full name
+    username,
+    firstName: fullname.split(' ')[0] || 'A',
+    lastName: fullname.split(' ')[1] || ' ',
+    discordUserId: data?.userInfo?.user?.discordUserId ?? '',
```

Comment:
Is there a need to check if these fields are ` null` or `undefined` using the nullish coalescing operator? Seems like unneeded additional complexity -- why not just use the logical OR operator? `||`

Reply:
Yeah, i think you're right about the added complexity but i get typescript errors when i use the `||` operator.
![Screen Shot 2022-04-14 at 20 27 10](https://user-images.githubusercontent.com/1410462/163462230-4f4bc325-42af-41b0-a3ce-e49dd28b45b8.png)

Reply:
maybe `_.get(data, 'userInfo.user.discordUserId', '')`

Reply:
@SlyBouhafs Thats because you're not using optional chaining. If you used optional chaining it should work. Or if you do what @songz suggested it should work